### DISCUSSION TITLE
Add mysql container

### DISCUSSION THREAD

**Discussion by d0niek** [nextcloud/all-in-one#1778] [Ideas] [answered]
Hi all!

What do you think about add a new container with `mysql` next to `postgresql`? Also add a new env to `nextcloud/all-in-one` like `DB_TYPE=pgsql/mysql`?

- Firstly administrators can choose a database which they know better ;-)
- Secondly if someone has already running self NC with `mysql` (like me) he can easily migrate to AIO.

You've describe migration process [here](https://github.com/nextcloud/all-in-one/blob/main/migration.md#migrate-the-files-and-the-database) which required (if needed) convert old DB to `postgresql`. Official documentations [said](https://docs.nextcloud.com/server/stable/admin_manual/configuration_database/db_conversion.html#converting-database-type) how to convert `sqlite` to `postgresql/mysql`. Maybe it support converts `mysql` to `postgresql` too? If yes, sorry for my post.

My idea is, safer is to allow users to run AIO with `mysql` or `postgresql` than required convert databases.

**Comment by szaimen** [comment, marked as answer]
> Maybe it support converts `mysql` to `postgresql` too?

yes, it does :)

**Reply by michaelfliegner** [reply]
Hi, I'm very interested. What became of this?

**Reply by nikbpetrov** [reply]
> Hi, I'm very interested. What became of this?

It's [still a nasty issue](https://github.com/nextcloud/server/issues/45257).
